imap/mboxevent.c

```diff
@@ -947,28 +963,15 @@ EXPORTED void mboxevent_extract_record(struct mboxevent *event, struct mailbox *
 
     /* add message EMAILID */
     if (mboxevent_expected_param(event->type, EVENT_MESSAGE_EMAILID)) {
-        char emailid[26];
-        emailid[0] = 'M';
-        memcpy(emailid+1, message_guid_encode(&record->guid), 24);
-        emailid[25] = '\0';
-        FILL_STRING_PARAM(event, EVENT_MESSAGE_EMAILID, xstrdup(emailid));
+        char *emailid = xmalloc(JMAP_EMAILID_SIZE);
+        jmap_set_emailid(&record->guid, emailid);
+        FILL_STRING_PARAM(event, EVENT_MESSAGE_EMAILID, emailid);
     }
 
     /* add message THREADID */
     if (mboxevent_expected_param(event->type, EVENT_MESSAGE_THREADID)) {
-        char threadid[18];
-        if (!record->cid) {
-            threadid[0] = 'N';
-            threadid[1] = 'I';
-            threadid[2] = 'L';
-            threadid[3] = '\0';
-        }
-        else {
-            threadid[0] = 'T';
-            memcpy(threadid+1, conversation_id_encode(record->cid), 16);
-            threadid[17] = '\0';
-        }
-        FILL_STRING_PARAM(event, EVENT_MESSAGE_THREADID, xstrdup(threadid));
+        FILL_STRING_PARAM(event, EVENT_MESSAGE_THREADID,
```

Comment:
Is it OK to remove the conditional that sets the `NIL` value here? I guess `set_threadid` sets a all-zeroes id if cid is zero.